hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsConfigHelper.java

```diff
@@ -0,0 +1,122 @@
+import static com.hazelcast.internal.metrics.ProbeLevel.INFO;
+
+public final class MetricsConfigHelper {
+    private static final ILogger LOGGER = Logger.getLogger(MetricsConfigHelper.class);
+
+    private MetricsConfigHelper() {
+    }
+
+    /**
+     * Overrides the {@link MetricsConfig} in the provided {@link Config}
+     * with the metrics system properties if present.
+     * See the {@link MetricsConfig} javadoc for the links between metrics
+     * configuration fields and the system properties.
+     *
+     * @param config The configuration
+     */
+    public static void overrideMetricsConfig(Config config) {
+        MetricsConfig metricsConfig = config.getMetricsConfig();
+        MetricsManagementCenterConfig managementCenterConfig = metricsConfig.getManagementCenterConfig();
+        MetricsJmxConfig jmxConfig = metricsConfig.getJmxConfig();
+
+        // MetricsConfig.enabled
+        tryOverride(config, GroupProperty.METRICS_ENABLED, prop -> metricsConfig.setEnabled(Boolean.parseBoolean(prop)),
+                () -> Boolean.toString(metricsConfig.isEnabled()), "MetricsConfig.enabled");
+
+        // MetricsManagementCenterConfig.enabled
+        tryOverride(config, GroupProperty.METRICS_MC_ENABLED,
+                prop -> managementCenterConfig.setEnabled(Boolean.parseBoolean(prop)),
+                () -> Boolean.toString(managementCenterConfig.isEnabled()), "MetricsManagementCenterConfig.enabled");
+
+        // MetricsManagementCenterConfig.retentionSeconds
+        tryOverride(config, GroupProperty.METRICS_MC_RETENTION,
+                prop -> managementCenterConfig.setRetentionSeconds(Integer.parseInt(prop)),
+                () -> Integer.toString(managementCenterConfig.getRetentionSeconds()),
+                "MetricsManagementCenterConfig.retentionSeconds");
+
+        // MetricsJmxConfig.enabled
+        tryOverride(config, GroupProperty.METRICS_JMX_ENABLED, prop -> jmxConfig.setEnabled(Boolean.parseBoolean(prop)),
+                () -> Boolean.toString(jmxConfig.isEnabled()), "MetricsJmxConfig.enabled");
+
+        // MetricsConfig.collectionFrequencySeconds
+        tryOverride(config, GroupProperty.METRICS_COLLECTION_FREQUENCY,
+                prop -> metricsConfig.setCollectionFrequencySeconds(Integer.parseInt(prop)),
+                () -> Integer.toString(metricsConfig.getCollectionFrequencySeconds()),
+                "MetricsConfig.collectionFrequencySeconds");
+    }
+
+    private static void tryOverride(Config hzConfig, HazelcastProperty property, Consumer<String> setterFn,
+                                    Supplier<String> getterFn, String configOverridden) {
+        String propertyValue = hzConfig.getProperty(property.getName());
+        try {
+            if (propertyValue != null) {
+                setterFn.accept(propertyValue);
+                LOGGER.warning(String.format("Overridden metrics configuration with system property '%s'='%s' -> '%s'='%s'",
```

Comment:
Does this log message need to have `warn` level? I'd say that it should have `info` level.

Reply:
Changed to `info`.